### PR TITLE
build: Eliminate need for x86-linux-gnu cross file

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Replace this with our own build-linux32.txt

Closes: #336

Signed-off-by: Joshua Ashton <joshua@froggi.es>

### CODE REVIEW

File: build-linux32.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+cpp = 'g++'
+ar = 'gcc-ar'
+strip = 'strip'
+
+[properties]
+c_args = ['-m32']
+c_link_args = ['-m32']
+cpp_args = ['-m32']
+cpp_link_args = ['-m32']
+pkg_config_path = ['/usr/lib32/pkgconfig']
```
Comment (DadSchoorse) [inline build-linux32.txt:12]: This isn't true on all distros.

**Comment by DadSchoorse** [comment]
Why not instruct people to use `CC="gcc -m32" CXX="g++ -m32" PKG_CONFIG="i686-pc-linux-gnu-pkg-config" meson ...`  with the correct `--libdir` for their distro instead of using a cross file?

**Comment by Joshua-Ashton** [comment]
Do we even need the package config stuff anymore? @HansKristian-Work

**Comment by Joshua-Ashton** [comment]
> Why not instruct people to use `CC="gcc -m32" CXX="g++ -m32" PKG_CONFIG="i686-pc-linux-gnu-pkg-config" meson ...` with the correct `--libdir` for their distro instead of using a cross file?

We could do that in the package-release script when you do ``--native`` perhaps and update the readme for it.

**Comment by HansKristian-Work** [comment]
@Joshua-Ashton I don't see a reason to remove the pkgconfig. If we're installing any .so/.a files we should also install a pkg-config for it.

File: build-linux32.txt (hunk 1 above)
Comment (HansKristian-Work) [reply build-linux32.txt:12]: For weird distros it should be fine to bring your own toolchain file or use good old envvars.

**Review by HansKristian-Work — APPROVED** [review]
LGTM.

**Pull request merged**